validator-samples/src/main/java/org/foo/modules/validators/TestNodeValidator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * ==========================================================================================
+ * =                            JAHIA'S ENTERPRISE DISTRIBUTION                             =
+ * ==========================================================================================
+ *
+ *                                  http://www.jahia.com
+ *
+ * JAHIA'S ENTERPRISE DISTRIBUTIONS LICENSING - IMPORTANT INFORMATION
+ * ==========================================================================================
+ *
+ *     Copyright (C) 2002-2020 Jahia Solutions Group. All rights reserved.
+ *
+ *     This file is part of a Jahia's Enterprise Distribution.
+ *
+ *     Jahia's Enterprise Distributions must be used in accordance with the terms
+ *     contained in the Jahia Solutions Group Terms &amp; Conditions as well as
+ *     the Jahia Sustainable Enterprise License (JSEL).
+ *
+ *     For questions regarding licensing, support, production usage...
+ *     please contact our team at sales@jahia.com or go to http://www.jahia.com/license.
+ *
+ * ==========================================================================================
+ */
+package org.foo.modules.validators;
+
+import org.jahia.services.content.JCRNodeWrapper;
+import org.jahia.services.content.decorator.validation.JCRNodeValidator;
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ *
+ * @author yousria
+ */
+@Component(immediate = true)
```

Comment:
sorry i missed that one, what is the `@Component` for ? validator are created on demand for each node to validate, i'm not even sure you can create a component with that ?

Reply:
thought it was mandatory but Ok I'll remove that